### ISSUE TITLE
[Suggestion] Color picker + Color favoriting

### ISSUE THREAD

**Issue by Timtee** [tsunamayo/Starship-EVO#5727]
I know in the past that there *was* talk about a color picker at one point or another, but... I'd personally like to suggest if at some point in one of the future updates, we can actually get such. Also on an a related note, the ability to favorite specific colors, as both could replace the current color palette solution near entirely practically.
An example of what I'm thinking though for the color picker in a sense:
![MCSS_298](https://github.com/tsunamayo/Starship-EVO/assets/38089299/ff0a62b9-77d0-4c89-a3e9-85698b147c1a)
Which yes, is of another unity game, Machinecraft's color picker but... It kinda is a decent enough example of what I'm thinking of And while yes the UI for this takes up an entire portion of the screen to the side, I figure that you could probably make a much more compact option within the Codex where the current color palette option already is, with the favorites being essentially just another section to the color palette menu below what is already there. 
Also to simplify the main color palette, is just to have the full on variant of each color, instead of each color with like, 2 shades darker or whatever below it. Mostly so that there is space for the color picker. Should it be squished into one portion of the ui instead of like, it's own little side menu that pops up to the side when accessing colors.
Also this could effectively move the materials into being part of the colors as well if utilizing a button or a slider to give a color reflectiveness like the materials.

TL:DR ; I'd just like to suggest that we get a color picker, and possibly even the option to favorite colors we use often on our builds.

**Comment by asanagisae** [comment]
I second this request, particularly the interaction with materials.

Dark steel and copper are some of my favorite "paints" due to their unique interactions with light. When you give a block an actual paint, it looks dull and cheap like plastic. A ship using only materials feels unfinished.

If we could have some degree of control over the specular and diffuse properties of colors, that'd be incredible. For example, to have a darker dark steel, or to produce a shiny emerald surface. If ambient, diffuse, and specular colors could be chosen independently, then we could even have pearlescent white with a hidden purple sheen when viewed from an angle. Imagine the possibilities!
